pkg/controller/machine.go

```diff
@@ -485,6 +499,21 @@ func (c *Controller) patchMachine(newMachine, oldMachine *machinev1alpha1.Machin
 		//nothing to do
 		return nil
 	}
+
+	// Kubernetes refuses requests with patch method if the payload contains a metadata field,
+	// shouldn't this get handled in jsonmergepatch?
+	var patchMap map[string]interface{}
+	err = json.Unmarshal(patch, &patchMap)
+	if err != nil {
+		fmt.Errorf("Failed to create a map from mergepatch: '%v'", err)
+	}
+	if _, ok := patchMap["metadata"]; ok {
+		delete(patchMap, "metadata")
+		patch, err = json.Marshal(patchMap)
+		if err != nil {
+			return fmt.Errorf("failed to marshal patch with removed metadata: '%v'", err)
+		}
+	}
```

Comment:
I've seen that some patches fail due to unstructured data send to the api.
Though i think this here will break the Finalizer handling.

I'm not sure whats the status on StrategicMergePatch and CRD's. I'll check

Reply:
It does indeed break the finalizer. Not sure what the expected behaviour is, but as long as the patch contained a metadata attribute with just a `resourceVersion` property, patching failed with a 
```
E0124 21:57:53.314219       1 status.go:64] apiserver received an error that is not an metav1.Status: unable to find api field in struct Unstructured for the json field "metadata"
```
from the api server.

Reply:
I see.
I'll test if we are safe to use update (put) instead